### ISSUE TITLE
Rewrite css21_dev/html4/{font-family,font}-013.htm to use a free font

### ISSUE THREAD

**Issue by SimonSapin** [servo/servo#8053]
These tests rely on some fonts from Microsoft whose license does not allow redistribution. Getting these fonts on our CI servers is proving troublesome: https://github.com/servo/servo/issues/7625#issuecomment-148613092

They should be modified to use different fonts whose license is not as problematic. To do so:

* Make sure the `msttcorefonts` package (or similar) is installed.
* Run `./mach test-css tests/wpt/css-tests/css21_dev/html4/font-013.htm tests/wpt/css-tests/css21_dev/html4/font-family-013.htm`. The tests should pass.
* Uninstall the package.
* Run the tests again. They should fail.
* In the test files, replace `"Times New Roman"`, `"Arial"`, and `"Courrier New"` with `serif`, `sans-serif`, and `monospace` respectively.
* Run the tests again, with the package still uninstalled. Hopefully they should start passing.
* Run `./mach run tests/wpt/css-tests/css21_dev/html4/font-family-013.htm` and `./mach run tests/wpt/css-tests/css21_dev/html4/fonts-013.htm` Check that each file renders shows the word "Ţęşţ" (Test with some diacritics on each letter) six times, and not for example just a square for each letter or nothing at all.

**Comment by metajack** [comment]
Isn't this really a bug that should be filed upstream?

cc @ms2ger @jgraham

**Comment by SimonSapin** [comment]
If the generic names `serif`, `sans-serif` or `monspace` don’t work, we’ll need to find some fonts that contain these characters and are available on all of our CI platforms.

**Comment by gsnedders** [comment]
@metajack yes, probably.

**Comment by SimonSapin** [comment]
I don’t know if this is really a bug upstream. When microsoft fonts are not available other browsers seem to pick other, similar fonts. The tests doesn’t really *require* non-free fonts, it’s just the easiest way we know to make it pass unmodified in our code. There’s probably more stuff we can improve in our font selection code.

(That said, if generic names work, there is no harm in making that change upstream too.)

**Comment by gsnedders** [comment]
Other browsers have magic to fallback for some super-common fonts. The test suite almost certainly shouldn't rely on non-free fonts or unspecified behaviour, IMO. I'm happy to champion (or help to) upstream (and I don't think there will be any disagreement).

**Comment by Ms2ger** [comment]
This change cannot be made here; it will be overwritten automatically.

**Comment by SimonSapin** [comment]
Good point, I’ve added a step to the instructions in the original message.

**Comment by 0X1A** [comment]
Willing to take this issue

**Comment by SimonSapin** [comment]
@0X1A, great! Feel free to ask questions here or on IRC (#servo on irc.mozilla.org).

**Comment by 0X1A** [comment]
@SimonSapin So, changing the fonts to their generic counterparts does not solve the tests and I get a fail for both.

Changing `reference/font-family-013-ref.html` to match the changes in `html4/font-family-013.html` gets the test to pass but `font-013.htm` still fails. Any suggestions? I'm guessing the ref's are upstream files?

**Comment by gsnedders** [comment]
@0X1A Do you have your changes anywhere?

**Comment by 0X1A** [comment]
@gsnedders Yes, the changes [are available here](https://github.com/0X1A/servo/commit/0525b1abcdafd6d253769b4c4814873c4e643d71)

**Comment by gsnedders** [comment]
That passes for me locally, FWIW.

**Comment by gsnedders** [comment]
As I just commented in #7625 I'm leaning towards there being some bug in the font-matching code?

**Comment by frewsxcv** [comment]
As per #9697, the switch to generic fonts fails on Linux, but passes on OSX

**Comment by frewsxcv** [comment]
Nevermind, I guess it fails on OSX on the builders too, despite passing locally on my OSX machine.

**Comment by jdm** [comment]
Turns out this wasn't easy.
